lib/generator-author.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+var nunjucks = require('nunjucks');
+var env = new nunjucks.Environment();
+var pathFn = require('path');
+var fs = require('fs');
+
+env.addFilter('uriencode', function(str) {
+  return encodeURI(str);
+});
+
+env.addFilter('noControlChars', function(str) {
+  return str.replace(/[\x00-\x1F\x7F]/g, '');
+});
+
+var atomTmplSrc = pathFn.join(__dirname, '../atom.xml');
+var atomTmpl = nunjucks.compile(fs.readFileSync(atomTmplSrc, 'utf8'), env);
+var rss2TmplSrc = pathFn.join(__dirname, '../rss2.xml');
+var rss2Tmpl = nunjucks.compile(fs.readFileSync(rss2TmplSrc, 'utf8'), env);
+
+module.exports = function(locals) {
+  var config = this.config;
+  var feedConfig = config.feed;
+  var template = feedConfig.type === 'rss2' ? rss2Tmpl : atomTmpl;
+
+  var posts = locals.posts.sort('-date');
+  posts = posts.filter(function(post) {
+    return post.draft !== true;
+  });
+
+  var url = config.url;
+  if (url[url.length - 1] !== '/') url += '/';
+
+  var authorIds = Object.keys(locals.data.authors);
+
+  var returnData = [];
+  for (var i = 0; i < authorIds.length; i++) {
```

Comment:
@stimms By the way, the reason it doesn't work is that `for( var authorId in authorIds )` gives you just the indexes of the author IDs, not the IDs themselves. And for whatever reason, you can't access authors by index. I.e. `locals.data.authors["simon_timms"]` works but `locals.data.authors[0]` doesn't. I haven't been able to figure out what kind of data structure the authors list is to detemrine why this is.